src/osd/PG.cc

```diff
@@ -3308,8 +3311,16 @@ void PG::reg_next_scrub()
   } else {
     scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
   }
-  if (is_primary())
+  if (is_primary()) {
+    if (!scrubber.must_scrub) {
+      // if not explicitly requested, postpone the scrub with a random delay
+      boost::rand48 rng;
+      boost::uniform_int<> time_range(g_conf->osd_scrub_min_interval,
```

Comment:
The scrubber.scrub_reg_stamp represents the last scrub (finishing) time for this PG, and will impact when the PG will be scheduled to scrub next time. With this implementation, it seems the scrubbing for next round will at least be postpone g_conf->osd_scrub_min_interval, and could be up to g_conf->osd_scrub_max_interval.

Does it make sense to just add a randomize value that is less than 1 hour (or make it configurable but default 1 hour) so that to respect the frequency of scrubbing as before?